File: rtix/ipc/tests/test_channel_map.cpp
// Copyright (c) 2025 Velodex Robotics, Inc and RTIX Developers.
// Licensed under Apache-2.0. http://www.apache.org/licenses/LICENSE-2.0

#include "rtix/ipc/tests/test_channel_map.h"
#include <gtest/gtest.h>
#include "rtix/ipc/channel_map.h"

using rtix::ipc::ChannelMap;

ChannelMap load_test_channel_map() {
  std::string yaml_file = "/rtix/rtix/ipc/tests/channel-map.yaml";
  YAML::Node yaml_node = YAML::LoadFile(yaml_file);
  return ChannelMap::LoadYaml(yaml_node);
}

TEST(Ipc, ChannelMap) {
  const auto channel_map = load_test_channel_map();
  EXPECT_EQ(channel_map.nodes.size(), 2);
  EXPECT_TRUE(channel_map.nodes.find("node_a") != channel_map.nodes.end());
  EXPECT_TRUE(channel_map.nodes.find("node_b") != channel_map.nodes.end());
}

File: rtix/ipc/tests/test_channel_map.py
# Copyright (c) 2025 Velodex Robotics, Inc and RTIX Developers.
# Licensed under Apache-2.0. http://www.apache.org/licenses/LICENSE-2.0

import yaml

from rtix.ipc.channel_map import ChannelMap

IPC_TEST_YAML_FILE = "/rtix/rtix/ipc/tests/channel-map.yaml"


def load_test_channel_map(yaml_file: str = IPC_TEST_YAML_FILE) -> ChannelMap:
    with open(yaml_file) as stream:
        yaml_dict = yaml.safe_load(stream)
        return ChannelMap.LoadYaml(yaml_dict=yaml_dict)


def test_channel_map():
    channel_map = load_test_channel_map()
    assert channel_map is not None
    assert len(channel_map.nodes) == 2
    assert "node_a" in channel_map.nodes
    assert "node_b" in channel_map.nodes
